OSDOCS-8914: Explaining upgrades for ROSA with HCP clusters. (#71393)
openshift/openshift-docs: modules/rosa-hcp-upgrading-cli-tutorial.adoc

====

.Prerequisites

* You have installed and configured the latest version of the ROSA CLI.

.Procedure

. Verify the current version of your cluster by running the following command:
+
[source,terminal]
----
$ rosa describe cluster --cluster=<cluster_name|cluster_id> <1>
----
<1> Replace `<cluster_name|cluster_id>` with the cluster name or the cluster ID.

. Verify that a control plane upgrade is available by running the following command:
+
[source,terminal]
----
$ rosa list upgrade --cluster=<cluster_name|cluster_id>
----
+
The command returns a list of upgradeable versions, including the recommended version.
+
.Example output
+
[source,terminal]
----
VERSION  NOTES
4.14.8   recommended
4.14.7
4.14.6
----

. Verify that a machine pool upgrade is available by running the following command:
+
[source,terminal]
----
$ rosa list upgrade --cluster <cluster-name> --machinepool <machinepool_name>
----
+
The command returns a list of upgradeable versions, including the recommended version.
+
.Example output
+
[source,terminal]
----
VERSION  NOTES
4.14.5   recommended
4.14.4
4.14.3
4.14.2
4.14.1

> ----

Reviewer: Just a feeling here but I find it confusing to show both `verify` commands one after the other and give an `upgrade` command options in the next bullet point.

+
[NOTE]
====
Your machine pools will not show upgrade versions beyond your cluster's current version. Ensure your cluster is up to date first.
====

. Upgrade your cluster with one of the following options:
** Upgrade the cluster's hosted control plane by running the following command:
+
[source,terminal]
----
$ rosa upgrade cluster -c <cluster_name> --control-plane [--schedule-date=XX --schedule-time=XX] [--version 4.14.8]
----
+
Your hosted control plane is now scheduled for an upgrade.

** Upgrade a specific machine pool on your cluster by running the following command:
+
[source,terminal]
----

> $ rosa upgrade machinepool -c <cluster_name> <your_machine_pool_id> [--version 4.14.8]

Reviewer: You can also have the `[--schedule-date=XX --schedule-time=XX]` option on upgrading a machinepool

// Module included in the following assemblies:
//
// * upgrading/rosa-hcp-upgrading.adoc

:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-upgrading-cli_{context}"]
= Upgrading with the ROSA CLI

You can manually upgrade a {hcp-title} cluster by using the ROSA CLI. This method schedules the cluster for an immediate upgrade if a more recent version is available.

[NOTE]
====

> Your control plane only supports machine pools within two minor Y-stream versions. For example, a {hcp-title} cluster with a control plane using version 4.15.z can support machine pools with version 4.13.z and 4.14.z, but the control plane will not support machine pools using version 4.12.z.

Reviewer 2: Suggestion to remove "can" since I think removing ambiguity in support statements might be clearer, but up to you:
```suggestion
Your control plane only supports machine pools within two minor Y-stream versions. For example, a {hcp-title} cluster with a control plane using version 4.15.z supports machine pools with version 4.13.z and 4.14.z, but the control plane will not support machine pools using version 4.12.z.
```

> Your control plane only supports machine pools within two minor Y-stream versions. For example, a {hcp-title} cluster with a control plane using version 4.15.z can support machine pools with version 4.13.z and 4.14.z, but the control plane will not support machine pools using version 4.12.z.

Reviewer 2: Suggestion for present tense:
```suggestion
Your control plane only supports machine pools within two minor Y-stream versions. For example, a {hcp-title} cluster with a control plane using version 4.15.z can support machine pools with version 4.13.z and 4.14.z, but the control plane does not support machine pools using version 4.12.z.
```

====

.Prerequisites

* You have installed and configured the latest version of the ROSA CLI.

.Procedure

. Verify the current version of your cluster by running the following command:
+
[source,terminal]
----

> $ rosa describe cluster --cluster=<cluster_name|cluster_id> <1>

Reviewer 2: For your consideration - the vertical column is probably fine but perhaps confusing to some, what do you think of something like this?
```suggestion
$ rosa describe cluster --cluster=<cluster_name_or_id> <1>
```
I think the callout label below can help describe it as an either/or situation, but up to you. If you do accept the suggestion, the same feedback applies to line 28.

----
<1> Replace `<cluster_name|cluster_id>` with the cluster name or the cluster ID.

> . List the versions that you can use to upgrade your control plane and machine pools by running the following command:

Reviewer 2: Wording suggestion:
```suggestion
. List the versions that you can upgrade your control plane and machine pools to by running the following command:
```

> . List the versions that you can use to upgrade your control plane and machine pools by running the following command:

Reviewer 2: Pluralize to account for multiple commands in the substeps:
```suggestion
. List the versions that you can use to upgrade your control plane and machine pools by running the following commands:
```

+
.. For the control plane versions, run the following command:
+
[source,terminal]
----
$ rosa list upgrade --cluster=<cluster_name|cluster_id>
----
+

> The command returns a list of upgradeable versions, including the recommended version.

Reviewer 2: I get what "upgradeable versions" is meant to say but I think it literally reads as the versions being upgradeable, rather than them being valid upgrade targets. In the enterprise update docs, I think we typically say "available updates" in these cases:
```suggestion
The command returns a list of available updates, including the recommended version.
```

+
.Example output
+
[source,terminal]
----
VERSION  NOTES
4.14.8   recommended
4.14.7
4.14.6
----
+
.. For the machine pool versions, run the following command:
+
[source,terminal]
----
$ rosa list upgrade --cluster <cluster-name> --machinepool <machinepool_name>
----
+

> The command returns a list of upgradeable versions, including the recommended version.

Reviewer 2: Same suggestion as above:
```suggestion
The command returns a list of available updates, including the recommended version.
```

+
.Example output
+
[source,terminal]
----
VERSION  NOTES
4.14.5   recommended
4.14.4
4.14.3
4.14.2
4.14.1
----
+
[NOTE]
====

> Your machine pools will not show upgrade versions beyond your cluster's current version. Ensure your cluster is up to date first.

Reviewer 2: Is this meant to say that the latest available update for machine pools is only as high as the current version of the control plane? If so, I think a rewording like this will be clearer to users:
```suggestion
The latest available update for machine pools is limited to the current version of the control plane. Ensure your control plane is up to date first.
```
I'm not familiar enough with HCP to know if "control plane" or "cluster" are interchangeable in this case, so this is only a suggestion.

====

. Upgrade your cluster with one of the following options:
** Upgrade the cluster's hosted control plane by running the following command:
+
[source,terminal]
----

> $ rosa upgrade cluster -c <cluster_name> --control-plane [--schedule-date=XX --schedule-time=XX] [--version 4.14.8]

Reviewer 2: I think the actual version number in this general command should be replaced with an input variable, otherwise if users followed this literally they'd always update their clusters to 4.14.8:
```suggestion
$ rosa upgrade cluster -c <cluster_name> --control-plane [--schedule-date=XX --schedule-time=XX] [--version <version_number>]
```

----
+
Your hosted control plane is now scheduled for an upgrade.

** Upgrade a specific machine pool on your cluster by running the following command:
+
[source,terminal]
----

> $ rosa upgrade machinepool -c <cluster_name> <your_machine_pool_id> [--schedule-date=XX --schedule-time=XX] [--version 4.14.8]

Reviewer 2: Same suggestion as above:
```suggestion
$ rosa upgrade machinepool -c <cluster_name> <your_machine_pool_id> [--schedule-date=XX --schedule-time=XX] [--version <version_number>]
```

----
+
Your machine pool is now scheduled for an upgrade.
//
// ** Upgrade both the hosted control plane and all attached machine pools by running the following command:
// +
// [source,terminal]
// ----

> // $ rosa upgrade cluster -c <cluster_name> [--schedule-date=XX --schedule-time=XX] [--version 4.14.8]

Reviewer 2: Since this is commented out this doesn't need to be fixed now, but if or when you add this command to the live docs, the same feedback about the version will apply:
```suggestion
// $ rosa upgrade cluster -c <cluster_name> [--schedule-date=XX --schedule-time=XX] [--version <version_number>]
```

// Module included in the following assemblies:
//
// * upgrading/rosa-hcp-upgrading.adoc

:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-upgrading-cli_{context}"]
= Upgrading with the ROSA CLI

You can upgrade a {hcp-title} (ROSA) cluster manually by using the ROSA CLI (`rosa`). This method schedules the cluster for an immediate upgrade, if a more recent version is available.

.Prerequisites

> * You have installed and configured the latest ROSA CLI on your installation host.

Reviewer 3: Should you use `rosa` now that you've explained the abbreviation a couple of sentences ago.

+
[source,terminal]
----
$ rosa list upgrade --cluster=<cluster_name|cluster_id>
----
+
The command returns a list of versions to which the cluster can be upgraded, including a recommended version.
+
.Example output
+
[source,terminal]
----
VERSION  NOTES
4.14.8   recommended
4.14.7
4.14.6
----

. To verify that a machinepool upgrade is available, enter the following command:
+
[source,terminal]
----
$ rosa list upgrade --cluster <cluster-name> --machinepool <machinepool_name>
----
+
The command returns a list of the versions to which your machinepools can be upgraded, including a recommended version.
+
.Example output
+
[source,terminal]
----
VERSION  NOTES
4.14.5   recommended
4.14.4
4.14.3
4.14.2
4.14.1
----
+
[NOTE]
====
Your machinepools will not show versions beyond your cluster's current version.
====

. You can upgrade your cluster's control plane or individual machine pools on your cluster.
** Run the following command to upgrade only the control plane of your {hcp-title} cluster.
+
[source,terminal]
----
$ rosa upgrade cluster -c <cluster_name> --control-plane [--schedule-date=XX --schedule-time=XX] [--version 4.14.8]
----
+

> Your control plane will be scheduled to be upgraded.

Reviewer 3: Should this be changed to ' Your control plane is scheduled to be upgraded' - as the scheduling has already taken place, right, and it's the upgrade that will be occurring in the future? Also can it be modified to let users know when it will be upgraded- depending on the --schedule-date and --schedule-time variable perhaps?

** Upgrade a specific machine pool on your cluster by running the following command:
+
[source,terminal]
----
$ rosa update machinepool -c <cluster_name> <your_machine_pool_id> [--version 4.14.8]
----
+
Your machine pool is now scheduled for an upgrade.
//
// ** Upgrade both the hosted control plane and all attached machine pools by running the following command:
// +
// [source,terminal]
// ----

> // $ rosa upgrade cluster -c <cluster_name> [--schedule-date=XX --schedule-time=XX] [--version 4.14.8]

Reviewer 4: as mentioned, this doesn't exist at the moment, I assume the // means commented out, so not appearing in the published doc right?

Author: Yup, I can totally remove it if we want to publish the rest of the updates until this is ready, but right now, it's totally hidden.

// ----
// +

> // Your control plane and machine pools will be scheduled to be upgraded.

Reviewer 3: Same comment as above.

.Troubleshooting

> * Sometimes a scheduled upgrade does not trigger. See link:https://access.redhat.com/solutions/6648291[Upgrade maintenance cancelled] for more information.

Reviewer 5: ```suggestion
* Sometimes a scheduled upgrade does not initiate. See link:https://access.redhat.com/solutions/6648291[Upgrade maintenance cancelled] for more information.
```
